app/controllers/api/v3/participants_controller.rb

```diff
@@ -55,8 +56,15 @@ def participants_params
         params.permit(:api_id, :sort, filter: %i[training_status from_participant_id])
       end
 
-      def withdraw_participant_params
-        params.require(:data).expect({ attributes: %i[course_identifier reason] })
+      def participant_action_params
```

Comment:
Happy to discuss/wait for other opinions, but I think this was clearer before; when its like this you have to dig into the service to figure out what it expects